lib/jekyll-paginate-v2/autopages/utils.rb

```diff
@@ -3,28 +3,53 @@ module PaginateV2::AutoPages
 
     class Utils
 
-      # Static: returns a fully formatted string with the tag macro (:tag) replaced
+      # Static: Expand placeholders within s
+      def self.expand_placeholders(s, placeholders)
+        # Create a pattern like /abc|def|./ for each key in placeholders
+        # Longer keys come first to ensure that a key like :foobar will take priority over :foo.
+        pattern = Regexp.new(((placeholders.keys.sort { |a, b| b.length <=> a.length }) + ["."]).join("|"))
+
+        # Format the output string. The pattern will cause scan() to return an array where each
+        # element is a single placeholder key, or a single character.
+        return s.scan(pattern).map { |token| placeholders.key?(token) ? placeholders[token] : token }.join
+      end
+
+      # Static: returns a fully formatted string with the tag macro (:tag) and tag name macro
+      # (:tag_name) replaced
       #
-      def self.format_tag_macro(toFormat, tag, slugify_config=nil)
+      def self.format_tag_macro(toFormat, tag, tag_name, slugify_config=nil)
         slugify_mode = slugify_config.has_key?('mode') ? slugify_config['mode'] : nil
         slugify_cased = slugify_config.has_key?('cased') ? slugify_config['cased'] : false
-        return toFormat.sub(':tag', Jekyll::Utils.slugify(tag.to_s, mode:slugify_mode, cased:slugify_cased))
+
+        return expand_placeholders(toFormat, {
+          ":tag"      => Jekyll::Utils.slugify(tag.to_s, mode:slugify_mode, cased:slugify_cased),
+          ":tag_name" => tag_name,
+        })
       end #function format_tag_macro
 
-      # Static: returns a fully formatted string with the category macro (:cat) replaced
+      # Static: returns a fully formatted string with the category macro (:cat) and category
+      # name macro (:cat_name) replaced
       #
-      def self.format_cat_macro(toFormat, category, slugify_config=nil)
+      def self.format_cat_macro(toFormat, category, category_name, slugify_config=nil)
         slugify_mode = slugify_config.has_key?('mode') ? slugify_config['mode'] : nil
         slugify_cased = slugify_config.has_key?('cased') ? slugify_config['cased'] : false
-        return toFormat.sub(':cat', Jekyll::Utils.slugify(category.to_s, mode:slugify_mode, cased:slugify_cased))
+
+        return expand_placeholders(toFormat, {
+          ":cat"      => Jekyll::Utils.slugify(category.to_s, mode:slugify_mode, cased:slugify_cased),
+          ":cat_name" => category_name,
```

Comment:
This line should call slugify to avoid invalid characters

Reply:
In the :cat_name expansion? The whole point is to be able to return the category name without being mangled - what characters are "invalid" here?

Reply:
I just gave this a bit of a try with the category `Tes<tin>g&"` and it comes out in the page as follows:

```
<title>Posts in category Tesg&amp;&quot; | ...</title>
<meta property="og:title" content="Posts in category Tesg&amp;&quot;" />
```

So quotes and ampersands are handled correctly, but anything between angle brackets is stripped out...

This is the same behaviour I get if I try putting angle brackets in a plain page title, e.g:

`title: 'foo"<bar>baz'`

in the page front matter comes out as:

`<title>foo&quot;baz | ...</title>`

An unmatched bracket in either (e.g. ` foo<bar`) comes out correctly (`foo&lt;bar`).

Any idea if this is a limitation/bug within Jekyll itself? Either way, it seems to be doing the right thing.